components/ord-service/src/main/java/com/sap/cloud/cmp/ord/service/controller/ODataController.java

```diff
@@ -38,17 +39,24 @@ public void handleODataRequest(HttpServletRequest request, HttpServletResponse r
 
     private JPAODataClaimsProvider createClaims(final HttpServletRequest request) throws IOException {
         final JPAODataClaimsProvider claims = new JPAODataClaimsProvider();
-
-        String tenantID = super.extractInternalTenantIdFromIDToken(request);
+        Pair<String, String> tenantIDs = super.extractTenantsFromIDToken(request);
+        if (tenantIDs == null) {
+            logger.warn("Could not determine tenants claim");
+            return claims;
+        }
+        String tenantID = tenantIDs.getFirst();
+        String providerTenantID = tenantIDs.getSecond();
         if (tenantID != null && !tenantID.isEmpty()) {
```

Comment:
Check the providerTenantID as well, as you expect it to be populated in all cases.

Reply:
I believe there is not a case in which we will receive **proivder_tenant_id** without **tenant_id**. In all other cases, if the tenant_id is present, the provider_id will also be present. We will either get it from the token or copy the tenant_id, so basically checking that tenant_id is present should be sufficient. However, I will change it if it looks hidden and odd.

Reply:
It will also return proper message in case of a bug in `extractTenantsFromIDToken`